### ISSUE TITLE
gui.show_dialog takes 13 px more space than it should on each side

### ISSUE THREAD

**Issue by ProditorMagnus** [wesnoth/wesnoth#8869]
### Game and System Information

- **Version:** 1.18.0

### Description of the bug

When using gui.show_dialog to show image to bottom left of screen, instead image is showed 13px up and 13px right from expected location. Image should be in green box. border_size/border can only be used to increase 13 to higher value.

### Steps to reproduce the behavior

```
# lua wesnoth.require("wml-utils").handle_event_commands(wml.load("~add-ons/EventLoader/action.cfg"))
[lua]
	code=<<
local T = wml.tag

function wesnoth.wml_actions.narration( cfg )
	local resolution = {
		T.helptip { id="tooltip_large" }, -- mandatory field
        T.tooltip { id="tooltip_large" }, -- mandatory field
        maximum_height = "(gamemap_height)",
        maximum_width = "(gamemap_width)",
        automatic_placement = true,
		vertical_placement="bottom",
		horizontal_placement="left",
        click_dismiss = true,
		border="all",
		border_size="0",
		T.grid {
			border="all",
			border_size="0",
			T.row {
				border="all",
				border_size="0",
				T.column {
					-- the only scope where border and border_size do anything when set above 0
					border="all",
					border_size="0",
					horizontal_alignment = "left",
					vertical_alignment = "bottom",
					T.image {
						id = "left_image",
						border="all",
						border_size="0"
					}
				}
			}
		}
	}
	
	local function narration_preshow(dialog)
		dialog:set_canvas(1,  {
			wml.tag.rectangle { x = 0, y = 0, w = "(width)", h = "(height)", fill_color= "0,255,0,200"},
			wml.tag.rectangle { x = 13, y = 13, w = "(width-26)", h = "(height-26)", fill_color= "255,0,0,200"},
		} )
		dialog.left_image.label = "portraits/humans/mage-arch.webp~O(1)"
	end
	local function narration_postshow(dialog)
	end
	gui.show_dialog( resolution, narration_preshow, narration_postshow )
end

wesnoth.wml_actions.narration{}
>>
[/lua]
```

### Additional context

![image](https://github.com/wesnoth/wesnoth/assets/7848963/100ff534-073f-416c-8ad4-68be7583b8d7)

**Comment by gfgtdf** [comment]
Is it documened anywhere that border_size="0" is supposed to work there? Otherwise this doesn't really sound like a bug to me.

**Comment by ProditorMagnus** [comment]
The fact that border_size works only in [column] was interesting, but not especially relevant. Inclusion of border_size just shows that it is not border that is causing the issue.

**Comment by gfgtdf** [comment]
> Inclusion of border_size just shows that it is not border that is causing the issue.

No, there a is cleary the dialogs border around the image, and if you cannot change that border via border_size then that doesn't make it a bug. The widgets (here: the window 'widget') definition decides how such widgets (here: the window 'widget') are drawn and laid out, and if you overwrite how it is drawn with dialog:set_canvas() then it doesn't magicially change how its laid out.

**Comment by ProditorMagnus** [comment]
Documentation has
> Every cell has a border_size and border the border_size is the number of pixels in the cell which aren't available for the widget. This is used to make sure the items in different cells aren't put side to side. With border it can be determined which sides get the border. So a border is either 0 or border_size.

Column documentation says border_size default is 0. Other tags do not say anything about borders.

**Comment by Vultraz** [comment]
Panels (of which windows are a type) accept `top_border`, `bottom_border`, `left_border`, and `right_border` in their [resolution[] tags. You should be able to make this issue go away by setting them all to zero. The values used by the default window definition are
```
left_border = 10
right_border = 13
top_border = 10
bottom_border = 13
```
so that would explain where the extra space is coming from. Though since they're excluded in your example, they *should* all default to zero... I don't know why they aren't in this case.

**Comment by ProditorMagnus** [comment]
> EDIT: It looks like the data provided to `show_dialog` is a window _instance_ config, and thus supports a `definition` key. Since you do not provide one, it defaults to the default window definition on which the border keys are set. If you want to get rid of the space, you can define a custom window definition, set the border values to zero, and use `show_dialog({ definition = "my_custom_definition" })`. It would also save you from having to override the canvas, since you could just define the canvas in the custom definition.

How to define custom window? I tried 
````
    gui.add_widget_definition("panel", "no_border_panel", {
        top_border = 0,
        bottom_border = 0,
        left_border = 0,
        right_border = 0
    })
```
But that started asking for id, description, [resolution] and so on. Not pleasant to test since after every attempt, game kicks me back to main menu.

**Comment by hladkyjakub** [comment]
I can confirm that indeed it can be easily done, but just for convenience, could mainline ship a borderless window definition.
[uninterestig, borderless window definition example](https://github.com/hladkyjakub/Bandits_from_Brown_Hills/blob/da48a2cc3ce075307f2d8411e097241214a444fe/gui/borderless_window.cfg)
To make it easier for beginners (back in days when this issue was created I struggled really bad with it)
